drivers/sensor/sensor_shell.c

```diff
@@ -113,6 +116,20 @@ enum dynamic_command_context {
 
 static enum dynamic_command_context current_cmd_ctx = NONE;
 
+/* Crate a single common config for one-shot reading */
+static enum sensor_channel iodev_sensor_shell_channels[SENSOR_CHAN_ALL];
+static struct sensor_read_config iodev_sensor_shell_read_config = {
+	.sensor = NULL,
+	.channels = iodev_sensor_shell_channels,
+	.num_channels = 0,
+	.max_channels = ARRAY_SIZE(iodev_sensor_shell_channels),
+};
+RTIO_IODEV_DEFINE(iodev_sensor_shell_read, &__sensor_iodev_api, &iodev_sensor_shell_read_config);
+
+/* Create the RTIO context to service the reading */
+RTIO_EXECUTOR_SIMPLE_DEFINE(sensor_shell_executor);
```

Comment:
Right, I want to add a new sensor shell command in the next PR `sensor stream icm42688 <trigger_0>..<trigger_N>` and since that one will be "always on" I thought we would want to use the concurrent executor. For the existing example the shell seems to be more of a bottleneck than the API so I went with what's simple to avoid overloading too many changes into the same PR.